File: ruby/hamming/hamming.rb
module Hamming
  VERSION = 2
  def self.compute(this, that)
    raise ArgumentError, "Nucleotides must be the same length" \
                         unless this.size == that.size
    this.size.times.count { |i| that[i] != this[i] }
  end
end

File: javascript/hamming/hamming.js
var Hamming = function() {};

Hamming.prototype.compute = function(a, b) {
  if (a.length != b.length) {
    throw "DNA strands must be of equal length.";
  }

  var distance = 0;
  for (var i = 0; i < a.length; i++) {
    if (a[i] != b[i]) {
        distance += 1;
    }
  }
  return distance;
};

module.exports = Hamming;
